src/eigenlayer/ZKMRStakeRegistry.sol

```diff
@@ -0,0 +1,227 @@
+    ) external onlyWhitelist(msg.sender) {
+        if (_isRegistered(msg.sender)) {
+            revert OperatorAlreadyRegistered();
+        }
+        totalOperators++;
+        operators[msg.sender] = publicKey;
+        serviceManager.registerOperatorToAVS(msg.sender, operatorSignature);
+        emit OperatorRegistered(msg.sender, address(serviceManager), publicKey);
+    }
+
+    function deregisterOperator() external {
+        if (!_isRegistered(msg.sender)) {
+            revert OperatorNotRegistered();
+        }
+        totalOperators--;
+        delete operators[msg.sender];
+        serviceManager.deregisterOperatorFromAVS(msg.sender);
+        emit OperatorDeregistered(msg.sender, address(serviceManager));
+    }
+
+    function updateOperatorKey(PublicKey calldata publicKey) external {
+        if (!_isRegistered(msg.sender)) {
+            revert OperatorNotRegistered();
+        }
+        operators[msg.sender] = publicKey;
+        emit OperatorUpdated(msg.sender, address(serviceManager), publicKey);
+    }
+
+    function updateQuorumConfig(Quorum memory quorum_) external onlyOwner {
+        _updateQuorumConfig(quorum_);
+    }
+
+    function updateMinimumWeight(uint256 newMinimumWeight) external onlyOwner {
+        uint256 oldMinimumWeight = minimumWeight;
+        minimumWeight = newMinimumWeight;
+        emit MinimumWeightUpdated(oldMinimumWeight, newMinimumWeight);
+    }
+
+    function quorum() external view returns (Quorum memory) {
+        return _quorum;
+    }
+
+    function isRegistered(address operator) external view returns (bool) {
+        return _isRegistered(operator);
+    }
+
+    function getOperatorShares(address operator)
+        external
+        view
+        returns (uint256)
+    {
+        return _getOperatorShares(operator);
+    }
+
+    function getOperatorWeight(address operator)
+        external
+        view
+        returns (uint256)
+    {
+        uint256 weight = _getOperatorShares(operator) / BPS;
+
+        if (weight >= minimumWeight) {
+            return weight;
+        } else {
+            return 0;
+        }
+    }
+
+    function _getOperatorShares(address operator)
+        private
+        view
+        returns (uint256)
+    {
+        uint256 totalShares;
+        StrategyParams[] memory strategyParams = _quorum.strategies;
+        IStrategy[] memory strategies = new IStrategy[](strategyParams.length);
+
+        for (uint256 i; i < strategyParams.length; i++) {
+            strategies[i] = strategyParams[i].strategy;
+        }
+
+        uint256[] memory shares =
+            delegationManager.getOperatorShares(operator, strategies);
+
+        for (uint256 i; i < strategyParams.length; i++) {
+            totalShares += shares[i] * strategyParams[i].multiplier;
+        }
+
+        return totalShares;
+    }
+
+    function _isRegistered(address operator) private view returns (bool) {
+        return operators[operator].x != 0;
```

Comment:
```suggestion
        return operators[operator].x != 0 || operators[operator].y != 0;
```
If initial key is `x == 0 && y != 0`, this function will return `false`. It can be a chance of double registering.

Reply:
Fixed by ensuring public key is valid on registration/update:
https://github.com/Lagrange-Labs/lagrange-lpn-contracts/pull/15/commits/5df4ea52c5ed910f7b55faa80c8cdaf8ac1a18e0